data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/TruncateStringProcessor.java

```diff
@@ -31,7 +32,11 @@ protected void performKeyAction(final Event recordEvent, final TruncateStringPro
         if (entry.getTruncateWhen() != null && !expressionEvaluator.evaluateConditional(entry.getTruncateWhen(), recordEvent)) {
             return;
         }
-        recordEvent.put(entry.getSource(), value.substring(0, entry.getLength()));
+        int startIndex = entry.getStartAt() == null ? 0 : entry.getStartAt();
+        Integer length = entry.getLength();
+        String truncatedValue = (length == null || startIndex+length >= value.length()) ? value.substring(startIndex) : value.substring(startIndex, startIndex + length);
```

Comment:
Should we catch `IndexOutOfBoundsException` so that this doesn't cause the pipeline to crash?